### PR TITLE
Add documentation to more WML unit tests.

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: data/test/scenarios/has_ally.cfg [hunk 1]
```diff
@@ -80,6 +101,20 @@
                 [/filter_side]
             [/have_unit]
         )}
+        {SUCCEED}
+    [/event]
+)}
+
+#####
+# API(s) being tested: [has_ally]
```
Comment (stevecotton) [inline data/test/scenarios/has_ally.cfg:109]: Also `[has_unit]`

File: data/test/scenarios/poison_opponent.cfg [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+# API(s) being tested: [poison]apply_to=opponent
+##
+# Actions:
+# Give both units the poison weapon special.
+# Have both units hit each other.
+##
+# Expected end state:
+# Both units are poisoned.
+#####
+
```
Comment (stevecotton) [inline data/test/scenarios/poison_opponent.cfg:13]: A general note - it seems inconsistent whether there's a blank line between the comment and the test or not. I'd prefer to omit this blank line from all the tests).

File: data/test/scenarios/has_ally.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/scenarios/has_ally.cfg]: Updated.

File: data/test/scenarios/poison_opponent.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/test/scenarios/poison_opponent.cfg]: Updated.

File: data/test/scenarios/interrupts.cfg [hunk 3]
```diff
@@ -102,6 +148,13 @@
     [/event]
 )}
 
+#####
+# API(s) being tested: [command][return]
+##
+# Expected end state:
+# [return] skips the entire rest of the event, not just the [command] it's nested in.
```
[the remark this answers is not in the document]
Comment (Pentarctagon) [reply data/test/scenarios/interrupts.cfg:150]: Created #7084

File: data/test/scenarios/interrupts.cfg [hunk 4]
```diff
@@ -19,6 +27,15 @@
...
 )}
 
+#####
+# API(s) being tested: [return]
+##
+# Actions:
+# Break out of an infinite loop with [return]
+##
+# Expected end state:
+# The loop exits.
```
Comment (stevecotton) [inline data/test/scenarios/interrupts.cfg:37]: Expectation is that the loop exits and the rest of the `[event]` is skipped. Note the `{FAIL}` after the `[/while]`.

File: data/test/scenarios/interrupts.cfg [hunk 5]
```diff
@@ -71,6 +103,15 @@
     [/event]
 )}
 
+#####
+# API(s) being tested: wesnoth.wml_actions.continue
+##
+# Actions:
+# Use continue directly in an event.
```
Comment (stevecotton) [inline data/test/scenarios/interrupts.cfg:110]: ..., outside a loop.

File: data/test/scenarios/put_to_recall_and_modify.cfg [hunk 6]
```diff
@@ -1,5 +1,9 @@
 #textdomain wesnoth-test
 
+# This is almost the same as the modify_unit_which_recall_list test, except that
+# creating Charlie on the map can leave an on-map x,y pair in Charlie's data,
+# which needs to be handled within the WML API.
```
Comment (stevecotton) [inline data/test/scenarios/put_to_recall_and_modify.cfg:5]: This no longer seems attached to the test that it's commenting on, I think it either needs to be accommodated in the new-style documentation or to be moved into the test's code.

File: data/test/scenarios/recall_by_unit_tag.cfg [hunk 7]
```diff
@@ -1,13 +1,19 @@
 # wmllint: no translatables
 
-# The [unit] tag can recall units if a unit with that id already exists
-# on the recall list. Some edge cases were unclear, so this test finds
-# out what they do.
-#
 # This is a test where any failure should probably be fixed by changing the
 # test to match changes in the engine. Having the test will be useful for
 # reference to find out when the engine behavior changed.
```
Comment (stevecotton) [inline data/test/scenarios/recall_by_unit_tag.cfg:5]: This is another where the comment is becoming detached from the test that it's documentation for.

File: data/test/scenarios/put_to_recall_and_modify.cfg (hunk 6 above)
Comment (Pentarctagon) [reply data/test/scenarios/put_to_recall_and_modify.cfg:5]: It's the only test in the file, so is there any confusion over which test it belongs to?

File: data/test/scenarios/interrupts.cfg (hunk 4 above)
Comment (Pentarctagon) [reply data/test/scenarios/interrupts.cfg:37]: Updated.

File: data/test/scenarios/interrupts.cfg (hunk 5 above)
Comment (Pentarctagon) [reply data/test/scenarios/interrupts.cfg]: Updated.

File: data/test/scenarios/put_to_recall_and_modify.cfg (hunk 6 above)
Comment (stevecotton) [reply data/test/scenarios/put_to_recall_and_modify.cfg:5]: Ah, I'd missed that detail. I agree, it's a lot clearer which test it belongs to in than case.

File: data/test/scenarios/recall_by_unit_tag.cfg (hunk 7 above)
Comment (stevecotton) [reply data/test/scenarios/recall_by_unit_tag.cfg:5]: Resolved as a non-issue because this is another where it's the only test in that file.

**Review by stevecotton — APPROVED** [review]

**Comment by stevecotton** [comment]
Please would you backport this to 1.16? It will make it easier to cherry-pick tests back to the stable branch.

Alternatively, I'd be happy to do it.

**Comment by Pentarctagon** [comment]
Is there really a benefit to it? If there's a test that needs to be cherry-picked back to the stable branch, it can just be copy-pasted.

**Comment by stevecotton** [comment]
I suppose it doesn't help - as you say, it's easy to just `git checkout` the whole file for the 1.17 version.
